build_stage_repository

```diff
@@ -136,6 +136,8 @@ def sync_prod_repository(collection, version, target_dir, dist, arch):
 
     cmd = [
         'dnf',
+        '--config',
+        f'tmp/dnf.conf',
```

Comment:
this should be `{base_dir}/dnf.conf` but we don't pass `base_dir` to those functions and I was lazy to make it so